Dockerfile

```diff
@@ -1,74 +1,137 @@
-
-# Product url constructs based on PRODUCT_NAME and PRODUCT_VERSION:
-#   https://download.jetbrains.com/idea/ideaIC-2020.2.3.tar.gz
-#   https://download.jetbrains.com/idea/ideaIU-2020.2.3.tar.gz
-#   https://download.jetbrains.com/webstorm/WebStorm-2020.2.3.tar.gz
-
-ARG BASE_MOUNT_FOLDER="/JetBrains"
+#
+# Copyright 2019-2020 JetBrains s.r.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
 
-# which is used by novnc to find websockify
-RUN yum install -y tigervnc-server supervisor wget java-11-openjdk-devel novnc fluxbox git which nss libXScrnSaver mesa-libgbm
+# Stage 1. Download IDE by downloadUrl build argument
+#   Requires only wget to download the IDE distributive.
+FROM alpine:3.13.4 as ideDownloader
+ARG downloadUrl
+WORKDIR /download
+RUN wget -q $downloadUrl -O - | tar -xz \
+    && find . -maxdepth 1 -type d -name * -exec mv {} /ide \;
 
-RUN mkdir /${PRODUCT_NAME}-${PRODUCT_VERSION} && \
-    case ${PRODUCT_NAME} in \
-        "ideaIC"|"ideaIU") \
-            wget -qO- https://download.jetbrains.com/idea/${PRODUCT_NAME}-${PRODUCT_VERSION}.tar.gz | tar -zxv --strip-components=1 -C /${PRODUCT_NAME}-${PRODUCT_VERSION} && \
-            ln -s /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.sh /opt/run-ide.sh \
-            ;; \
-        "WebStorm") \
-            wget -qO- https://download.jetbrains.com/webstorm/${PRODUCT_NAME}-${PRODUCT_VERSION}.tar.gz | tar -zxv --strip-components=1 -C /${PRODUCT_NAME}-${PRODUCT_VERSION} && \
-            ln -s /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/webstorm.sh /opt/run-ide.sh \
-            ;; \
-    esac && \
-    mkdir -p ${BASE_MOUNT_FOLDER}/${PRODUCT_NAME} && \
-    mkdir /etc/default/jetbrains && \
-    for f in "${BASE_MOUNT_FOLDER}" "/${PRODUCT_NAME}-${PRODUCT_VERSION}" "/etc/passwd" "/etc/default/jetbrains"; do \
-      echo "Changing permissions on ${f}" && chgrp -R 0 ${f} && \
-      chmod -R g+rwX ${f}; \
-    done && \
-    echo "idea.config.path=${BASE_MOUNT_FOLDER}/${PRODUCT_NAME}/config" > /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.properties && \
-    echo "idea.system.path=${BASE_MOUNT_FOLDER}/${PRODUCT_NAME}/caches" >> /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.properties && \
-    echo "idea.plugins.path=${BASE_MOUNT_FOLDER}/${PRODUCT_NAME}/plugins" >> /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.properties && \
-    echo "idea.log.path=${BASE_MOUNT_FOLDER}/${PRODUCT_NAME}/logs" >> /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.properties
+# Stage 2. Build Projector
```

Comment:
Could these three build steps be combined into a single container image ? I'm not sure but I think Brew doesn't like multi-stage builds when multi = 3 or more. (Ideally, we'd have one build stage, one runtime stage.)

Reply:
Combined step 1 and 3 into a single step. As for second one, let it be as is, if it is possible. First stage builds using gradle projector itself. Second stage repacks IDE packaging with built projector assembly. Third stage builds runtime image.